src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc

```diff
@@ -65,23 +67,53 @@ void GcsAutoscalerStateManager::HandleReportAutoscalingState(
     rpc::SendReplyCallback send_reply_callback) {
   RAY_CHECK(thread_checker_.IsOnSameThread());
 
-  // Never seen any autoscaling state before - so just takes this.
-  if (!autoscaling_state_.has_value()) {
-    autoscaling_state_ = *std::move(request.mutable_autoscaling_state());
-    send_reply_callback(ray::Status::OK(), nullptr, nullptr);
-    return;
-  }
-
-  // Cancel the infeasible requests if the feature is enabled
-  std::function<void()> callback = [this]() {
+  // Create the callback to cancel the infeasible requests if the feature is enabled
+  bool has_new_infeasible_requests = false;
+  std::function<void()> callback = [this, &has_new_infeasible_requests]() {
     bool enable_infeasible_task_early_exit =
         RayConfig::instance().enable_infeasible_task_early_exit();
 
     if (enable_infeasible_task_early_exit) {
       this->CancelInfeasibleRequests();
+    } else if (has_new_infeasible_requests) {
+      // publish error message
+      std::string error_message =
+          "There are tasks with infeasible resource requests and won't "
+          "be scheduled. See "
+          "https://docs.ray.io/en/latest/ray-core/scheduling/"
+          "index.html#ray-scheduling-resources "
+          "for more details. Please consider taking the following actions to solve "
+          "the issue: "
+          "1. Updating the ray cluster to include nodes with all required resources "
+          "in order to let the tasks be scheduled. 2. To prevent the tasks from "
+          "hanging, you can consider enabling the infeasible task early exit feature "
+          "by setting the 'RAY_enable_infeasible_task_early_exit' env var to 'true'."
+          "In a future release of Ray, we are planning to enable infeasible task "
+          "early exit by default.";
```

Comment:
```suggestion
          "1. Update the ray cluster to include nodes with all required resources. "
          "2. To cause the tasks to raise an error instead of hanging, set 'RAY_enable_infeasible_task_early_exit=true'. This feature will be turned on by default in a future version of Ray.";
```

Reply:
Suggestions to make it more concise. No need to add flowery language like "please consider" etc.